### ISSUE TITLE
Reverse Proxy Terminated SSL appears not to work.

### ISSUE THREAD

**Issue by iteratee** [nuxsmin/sysPass#1741]
**sysPass Version**

3.2.2

**Describe the bug**

Syspass appears not to work behind SSL-terminating reverse proxy.

**To Reproduce**
Run docker-compose sysPass as follows:
sudo docker-compose -p syspass -f docker-compose.yml up -d
docker-compose.yml contents:

```
version: '2'
services:
  app:
    container_name: syspass-app
    image: syspass/syspass:3.2.2
    restart: always
    ports:
      - "127.0.0.1:51500:80"
      - "127.0.0.1:51501:443"
    links:
      - db
    volumes:
      - syspass-config:/var/www/html/sysPass/app/config
      - syspass-backup:/var/www/html/sysPass/app/backup
    environment:
      - USE_SSL=no
  db:
    container_name: syspass-db
    restart: always
    image: mariadb:10.2
    environment:
      - MYSQL_ROOT_PASSWORD=syspass
    ports:
      - "3306"
    volumes:
      - syspass-db:/var/lib/mysql

volumes:
  syspass-config: {}
  syspass-backup: {}
  syspass-db: {}
```
nginx-config for site:
```
server {
	listen 80;
	listen [::]:80;
	server_name syspass.iteratee.net;

	root /var/www/syspass.iteratee.net;

	location /.well-known/acme-challenge {
		# First attempt to serve request as file, then
		# as directory, then fall back to displaying a 404.
		try_files $uri $uri/ =404;
	}
	location / {
		proxy_pass http://localhost:51500;
                proxy_set_header X-Real-IP         $remote_addr;
                proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Forwarded-Host  $host;
                proxy_set_header Host              $host;
                proxy_set_header X-Forwarded-Port  $server_port;
	}

	# Add index.php to the list if you are using PHP
	index index.php index.html index.htm index.nginx-debian.html;
}
server {
	listen 443 ssl;
	listen [::]:443 ssl;
	server_name syspass.iteratee.net;

	root /var/www/syspass.iteratee.net;

	location /.well-known/acme-challenge {
		# First attempt to serve request as file, then
		# as directory, then fall back to displaying a 404.
		try_files $uri $uri/ =404;
	}
	location / {
		proxy_pass http://localhost:51501;
                proxy_set_header X-Real-IP         $remote_addr;
                proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Forwarded-Host  $host;
                proxy_set_header Host              $host;
                proxy_set_header X-Forwarded-Port  $server_port;
	}

	# Add index.php to the list if you are using PHP
	index index.php index.html index.htm index.nginx-debian.html;
}
```
**Expected behavior**
There should be some environment variable that causes syspass-app to honor X-Forwarded-Proto

**Screenshots**
![Screenshot](https://user-images.githubusercontent.com/4795900/130529852-b15a9b14-4bc6-4790-9b16-7f3c3b58ba5d.jpg)

**Event log**

```
syspass-app | 172.24.0.1 - - [23/Aug/2021:22:41:36 +0000] "GET / HTTP/1.0" 302 532 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.42 Safari/537.36"
syspass-app | [Mon Aug 23 22:41:36.819130 2021] [php7:notice] [pid 591] [client 172.24.0.1:38192] [INFO] [Extensions checked] SP\\Core\\PhpExtensionChecker::checkMandatory
syspass-app | [Mon Aug 23 22:41:36.822918 2021] [php7:notice] [pid 591] [client 172.24.0.1:38192] [INFO] [Saved icons cache] SP\\Core\\UI\\Theme::saveIcons
syspass-app | [Mon Aug 23 22:41:36.823671 2021] [php7:notice] [pid 591] [client 172.24.0.1:38192] [ERROR] [Not installed] SP\\Modules\\Web\\Init::initialize
syspass-app | [Mon Aug 23 22:41:36.825238 2021] [php7:notice] [pid 591] [client 172.24.0.1:38192] [INFO] [Loaded actions cache] SP\\Core\\Acl\\Actions::loadCache
syspass-app | [Mon Aug 23 22:41:36.826274 2021] [php7:notice] [pid 591] [client 172.24.0.1:38192] [EXCEPTION] [Response is locked\n#0 /var/www/html/sysPass/vendor/klein/klein/src/Klein/AbstractResponse.php(211): Klein\\AbstractResponse->requireUnlocked()\n#1 /var/www/html/sysPass/vendor/klein/klein/src/Klein/AbstractResponse.php(523): Klein\\AbstractResponse->code(Integer)\n#2 /var/www/html/sysPass/app/modules/web/Controllers/IndexController.php(50): Klein\\AbstractResponse->redirect(String)\n#3 [internal function]: SP\\Modules\\Web\\Controllers\\IndexController->indexAction()\n#4 /var/www/html/sysPass/lib/SP/Bootstrap.php(240): call_user_func_array(Array,Array)\n#5 [internal function]: SP\\Bootstrap->SP\\{closure}(Object(Klein\\Request),Object(Klein\\Response),Object(Klein\\ServiceProvider),Object(Klein\\App),Object(Klein\\Klein),Object(Klein\\DataCollection\\RouteCollection),Array)\n#6 /var/www/html/sysPass/vendor/klein/klein/src/Klein/Klein.php(879): call_user_func(Object(Closure),Object(Klein\\Request),Object(Klein\\Response),Object(Klein\\ServiceProvider),Object(Klein\\App),Object(Klein\\Klein),Object(Klein\\DataCollection\\RouteCollection),Array)\n#7 /var/www/html/sysPass/vendor/klein/klein/src/Klein/Klein.php(588): Klein\\Klein->handleRouteCallback(Object(Klein\\Route),Object(Klein\\DataCollection\\RouteCollection),Array)\n#8 /var/www/html/sysPass/lib/SP/Bootstrap.php(464): Klein\\Klein->dispatch(Object(Klein\\Request))\n#9 /var/www/html/sysPass/lib/Base.php(75): SP\\Bootstrap->run(Object(DI\\Container))\n#10 /var/www/html/sysPass/index.php(28): require(String)] N/A
syspass-app | [Mon Aug 23 22:41:36.826632 2021] [php7:notice] [pid 591] [client 172.24.0.1:38192] [EXCEPTION] [Response is locked\n#0 /var/www/html/sysPass/vendor/klein/klein/src/Klein/AbstractResponse.php(157): Klein\\AbstractResponse->requireUnlocked()\n#1 /var/www/html/sysPass/lib/SP/Bootstrap.php(149): Klein\\AbstractResponse->body(String)\n#2 [internal function]: SP\\Bootstrap->SP\\{closure}(Object(Klein\\Klein),String,String,Object(Klein\\Exceptions\\LockedResponseException))\n#3 /var/www/html/sysPass/vendor/klein/klein/src/Klein/Klein.php(936): call_user_func(Object(Closure),Object(Klein\\Klein),String,String,Object(Klein\\Exceptions\\LockedResponseException))\n#4 /var/www/html/sysPass/vendor/klein/klein/src/Klein/Klein.php(645): Klein\\Klein->error(Object(Klein\\Exceptions\\LockedResponseException))\n#5 /var/www/html/sysPass/lib/SP/Bootstrap.php(464): Klein\\Klein->dispatch(Object(Klein\\Request))\n#6 /var/www/html/sysPass/lib/Base.php(75): SP\\Bootstrap->run(Object(DI\\Container))\n#7 /var/www/html/sysPass/index.php(28): require(String)] N/A
syspass-app | 172.24.0.1 - - [23/Aug/2021:22:41:36 +0000] "GET /index.php HTTP/1.0" 302 335 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.42 Safari/537.36"
syspass-app | [Mon Aug 23 22:41:37.312813 2021] [php7:notice] [pid 588] [client 172.24.0.1:38198] [INFO] [Extensions checked] SP\\Core\\PhpExtensionChecker::checkMandatory
syspass-app | [Mon Aug 23 22:41:37.314490 2021] [php7:notice] [pid 588] [client 172.24.0.1:38198] [INFO] [Saved icons cache] SP\\Core\\UI\\Theme::saveIcons
syspass-app | [Mon Aug 23 22:41:37.315595 2021] [php7:notice] [pid 588] [client 172.24.0.1:38198] [INFO] [Loaded actions cache] SP\\Core\\Acl\\Actions::loadCache
syspass-app | 172.24.0.1 - - [23/Aug/2021:22:41:37 +0000] "GET /index.php?r=install/index HTTP/1.0" 200 3575 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.42 Safari/537.36"
```

**Platform (please complete the following information):**
 - OS: Linux (using docker)
 - OS Version: Ubuntu 20.04
 - Browser Chrome

**Additional context**
I don't know if the Klein Errors are related or not.

**Comment by nuxsmin** [comment]
Hello,

it seems you need to set the application URL in order to use the correct URL when making the requests. Please be aware that sysPass doesn't know the browser protocol when it's behind a reverse proxy, so the only way is to force to use a fixed URL.

Regards.

**Comment by loutremaline** [comment]
Hello,

I've done a fresh install of the lastest sysPass version (3.2.11) with docker image (no docker compose) behind a reverse proxy. Reverse proxy serves the secure connection (HTTPS) with clients. And I encountered the same problem with sysPass for loading resources. I've found that http headers added by reverse proxy are not or no longer correctly managed by sysPass. Then sysPass creates links for resources without https, ignoring headers that specify incoming https request. In particular, there is an error in the construction of the host name regex. But not only.

@nuxsmin Before declaring a bug, I looked to see if there was already one on this issue. And I found this ticket. I attach my diff file, which is very short, that allowed me to correct the problem (only for using the "Forwarded" header). Look at my (short) patch file for code fix. [Request.patch.txt](https://github.com/nuxsmin/sysPass/files/13863058/Request.patch.txt)
